### PR TITLE
Course browser works but still waiting on api endpoint to completely finish it. 

### PR DESCRIPTION
**Pull request by eyadsw**

### CODE REVIEW

File: App/src/pages/CourseBrowser.tsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 import { Box, Button, Container, FormLabel, Heading, Stack } from "@chakra-ui/react";
-
+import { useNavigate } from "react-router-dom";
+import { Input } from "@chakra-ui/react";
 interface SubjectItem {
     value: string;
     label: string;
 }
 
+// when the api is implemented I will make the subjects available in the drop down
```
Comment (JamesPartsafas) [inline App/src/pages/CourseBrowser.tsx:11]: What API endpoint are you waiting for? An endpoint to get all course subjects? This exists, it's `/Course/GetAllCourseSettings`, it's already being used by the add-course page

File: App/src/pages/CourseBrowser.tsx [hunk 2]
```diff
@@ -18,13 +20,22 @@ interface CatalogItem {
...
-const CATALOG_ITEMS: Array<CatalogItem> = [
-    { value: "201", label: "201" },
-    { value: "301", label: "301" },
-    { value: "401", label: "401" },
-];
+const CourseSelector1: React.FC<{ onChange: (selectedOption: SubjectItem | null) => void }> = ({ onChange }) => (
+    <Select options={SUBJECT_ITEMS} onChange={onChange} />
+);
+
+function App() {
```
Comment (JamesPartsafas) [inline App/src/pages/CourseBrowser.tsx:27]: Can you rename this function from App to CourseBrowser to match the file name?

**Comment by JamesPartsafas** [comment]
Can you pull from main to fix the merge conflict? Also can you create an issue for this task, linked to a parent story as we do for other tasks?

File: App/src/pages/CourseDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+const CourseDetails = () => {
+    const query = useQuery(); // Use the useQuery function
+
+    // Access specific query parameters
+    const subject: string = query.get("subject");
+    const catalogNumber: string = query.get("catalog");
+    const [data, setData] = useState([]);
+
+    useEffect(() => {
+        fetch("/data.json")
```
Comment (JamesPartsafas) [inline App/src/pages/CourseDetails.tsx:18]: I don't quite understand what you're fetching here. Is this a case of not having an endpoint available to use? If that's the case, you can make a relevant endpoint as part of the task or as a separate task.

File: App/src/pages/CourseBrowser.tsx [hunk 4]
```diff
@@ -1,12 +1,14 @@
-import React from "react";
+import React, { useState } from "react";
```
Comment (JamesPartsafas) [inline App/src/pages/CourseBrowser.tsx:1]: The styling in the browser is good.

**Comment by JamesPartsafas** [comment]
There seem to be 2 separate tasks here, one for the course browser which is just about ready to go after linking to the relevant API endpoint, and a course details page that seems a bit further off. If you want you can separate this into separate PRs, so that the working course browser can be merged while the course details page continues being worked on.

**Comment by AgnesCroteau** [comment]
I had originally deleted that branch, and I only restored it to find out what bug happened after merging (as discussed today).  Sounds like I should have recreated a new branch?

**Comment by JamesPartsafas** [comment]
> I had originally deleted that branch, and I only restored it to find out what bug happened after merging (as discussed today). Sounds like I should have recreated a new branch?

Whether it's on the same branch or not doesn't really matter (but it's generally better to have a separate one for each task), what's more important is 1 PR per issue, so that things can get merged incrementally and everybody can stay up to date with other people's work (and of course this an expectation in our grading scheme). In this case there's not actually any issue at all this PR maps to. If you look at the general pattern followed in this repo, there's a parent story with multiple child tasks. For each child task, there's a PR associated to them to close them. Additionally, there's references to this child task in each commit message. For example, you have the parent story #7 with many child tasks, such as #61 and this child task is closed by PR #62 . In each commit message for that PR, you can see a reference to the child task.

**Comment by AgnesCroteau** [comment]
> > I had originally deleted that branch, and I only restored it to find out what bug happened after merging (as discussed today). Sounds like I should have recreated a new branch?
> 
> Whether it's on the same branch or not doesn't really matter (but it's generally better to have a separate one for each task), what's more important is 1 PR per issue, so that things can get merged incrementally and everybody can stay up to date with other people's work (and of course this an expectation in our grading scheme). In this case there's not actually any issue at all this PR maps to. If you look at the general pattern followed in this repo, there's a parent story with multiple child tasks. For each child task, there's a PR associated to them to close them. Additionally, there's references to this child task in each commit message. For example, you have the parent story #7 with many child tasks, such as #61 and this child task is closed by PR #62 . In each commit message for that PR, you can see a reference to the child task.

It's a bit silly but I'm not quite sure how to approach this at the moment especially since the branch has now conflicts that are already resolved per my previous PR, at least to my understanding.

**Comment by JamesPartsafas** [comment]
Duplicate of #230

**Pull request closed without merging**